pkg/controllers/workgenerator/controller.go

```diff
@@ -659,10 +659,14 @@ func (r *Reconciler) SetupWithManager(mgr controllerruntime.Manager) error {
 				}
 				oldAppliedStatus := meta.FindStatusCondition(oldWork.Status.Conditions, fleetv1beta1.WorkConditionTypeApplied)
 				newAppliedStatus := meta.FindStatusCondition(newWork.Status.Conditions, fleetv1beta1.WorkConditionTypeApplied)
-				// we only need to handle the case the applied condition is flipped between true and NOT true between the
+				oldAvailableStatus := meta.FindStatusCondition(oldWork.Status.Conditions, fleetv1beta1.WorkConditionTypeAvailable)
+				newAvailableStatus := meta.FindStatusCondition(newWork.Status.Conditions, fleetv1beta1.WorkConditionTypeAvailable)
+
+				// we only need to handle the case the applied or available conditions is flipped between true and NOT true between the
 				// new and old work objects. Otherwise, it won't affect the binding applied condition
-				if condition.IsConditionStatusTrue(oldAppliedStatus, oldWork.GetGeneration()) == condition.IsConditionStatusTrue(newAppliedStatus, newWork.GetGeneration()) {
-					klog.V(2).InfoS("The work applied condition didn't flip between true and false, no need to reconcile", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
+				if condition.IsConditionStatusTrue(oldAppliedStatus, oldWork.GetGeneration()) == condition.IsConditionStatusTrue(newAppliedStatus, newWork.GetGeneration()) &&
+					condition.IsConditionStatusTrue(oldAvailableStatus, oldWork.GetGeneration()) == condition.IsConditionStatusTrue(newAvailableStatus, newWork.GetGeneration()) {
```

Comment:
Should it be if there are no changes between old condition and new condition then we'll skip the reconcile?

eg, oldWork applied condition is false and newWork applied condition is false,

We could use the EqualCondition func to compare https://github.com/Azure/fleet/blob/main/pkg/utils/condition/condition.go#L16 

current = old, desired = new // though old observed generation should never greater than the new one.